src/pages/analytics/Analytics.js

```diff
@@ -192,54 +189,54 @@ class Analytics extends Page {
         this.context.updateAppState({
             showSnackbar: !completed,
             pageState: {
-                notifications,
+                notifications: this.getUpdatedNotifications(taskSummary),
                 loading: !completed,
             },
         })
     }
 
-    verifyInProgressJobsForTaskSummaryResponseAndUpdateState = taskSummaryResponse => {
-        const jobIds = taskSummaryResponse
-            ? Object.keys(taskSummaryResponse)
-            : []
-
-        // looking for the most recent in progress job
-        for (let i = jobIds.length - 1; i >= 0; i--) {
-            const jobId = jobIds[i]
-            const notifications = taskSummaryResponse[jobId] || []
-
-            // found in progress job: show current notifications and starts pooling
-            if (this.isJobInProgress(notifications)) {
-                const intervalId = this.startsPooling()
-
-                this.context.updateAppState({
-                    showSnackbar: true,
-                    snackbarConf: {
-                        type: LOADING,
-                    },
-                    pageState: {
-                        notifications,
-                        loading: true,
-                        jobId,
-                        intervalId,
-                    },
-                })
+    getUpdatedNotifications(taskSummary = []) {
+        // Notification table needs to be updated when new tasks are added
+        if (taskSummary.length <= this.state.notifications.length) {
+            return this.state.notifications
+        }
 
-                break
-            }
+        const lastIndex = taskSummary.length - 1
+
+        // Reverse to sort oldest-newest
+        // Assumption: all tasks are completed, apart from the last one,
+        // which is the in-progress task.
+        // Exception is when the most recent task comes back as completed
+        // this indicates the entire process is done, so we respect that completed status.
+        return taskSummary.reverse().map((x, i) => ({
+            ...x,
+            completed: x.completed || i < lastIndex,
+        }))
+    }
+
+    verifyInProgressJobsForTaskSummaryResponseAndUpdateState = taskSummaryResponse => {
```

Comment:
This method name gives me java flashbacks. Not that you can do anything about this, but I feel triggered.

Reply:
Well actually, would it not make sense to break this somehow? The `And` leads me to believe that it could be two methods. Might make things more clear, because this method name is a bit much.

Reply:
You're going off of the existing code of course. So we could also deal with this in a follow up PR.

Reply:
> You're going off of the existing code of course. So we could also deal with this in a follow up PR.

Let's do that